### PR TITLE
AvatarWithBadge Housekeeping

### LINKED ISSUE #16179: 🧹 [UI House Keeping] AvatarWithBadge
### Description
Ensure that `AvatarWithBadge` adheres to all of the following conventions and standards

- [ ] Has a `className` prop and the PropType descriptions are all the same
- [ ] Prop table in MDX docs have the "Accepts all Box component props" description and link
- [ ] We are consistent when using the same prop names like `size` and are suggesting the use of the generalized `design-system.js` constants e.g. `SIZES` as the primary option but noting the component consts in the documentation and using them for propType validation and storybook controls only
- [ ] Standardize all similar prop names for images `imgSrc`, `imgAlt`(html element + attribute) (needs audit)
- [ ] We have a story for each component prop and we use the prop name verbatim e.g. `size` prop would be `export const Size = (args) => (`
- [ ] We have the accompanying documentation for each component prop and we use the prop name verbatim e.g. `size` prop would be `### Size`
- [ ] Are multiple props stories allowed? e.g. `Color, Background Color And Border Color` story in `base-avatar` - [ ] yes when it makes sense to
- [ ] All Base components follow the suffix convention e.g. `ButtonBase`
- [ ] All Base component MDX documentation have the base component notification at the top
- [ ] Add `mm-` prefix to all classNames
- [ ] className is kebab case version of the component name
- [ ] Spread base components props and reduce duplication of props when props aren't being changed and remain the same for both variant and base components
- [ ] Add component to root `index.js` file in component-library
- [ ] Add locals for any default text I18nContext as default context 
- [ ] Add any "to dos" with a `// TODO:` comment so we can search for them at a later date e.g. blocking components etc
- [ ] Add snapshot testing
- [ ] Add pixel values to propType descriptions if we use abstracted prop types that relate to pixel values e.g. `SIZE.MD (32px)`
- [ ] Each prop section in the MDX docs should have: a heading, a description, a story and an example code snipped

### PR DESCRIPTION
**Pull request by NidhiKJha**
* Fixes #16179 

This PR is to ensure that `AvatarWithBadge` adheres to all of the following conventions and standards

- [x] Has a `className` prop and the PropType descriptions are all the same
- [x] Prop table in MDX docs have the "Accepts all Box component props" description and link
- [x] We are consistent when using the same prop names like `size` and are suggesting the use of the generalized `design-system.js` constants e.g. `SIZES` as the primary option but noting the component consts in the documentation and using them for propType validation and storybook controls only
- [ ] Standardize all similar prop names for images `imgSrc`, `imgAlt`(html element + attribute) (needs audit)
- [x] We have a story for each component prop and we use the prop name verbatim e.g. `size` prop would be `export const Size = (args) => (`
- [ ] We have the accompanying documentation for each component prop and we use the prop name verbatim e.g. `size` prop would be `### Size`
- [ ] Are multiple props stories allowed? e.g. `Color, Background Color And Border Color` story in `base-avatar` - [ ] yes when it makes sense to
- [ ] All Base components follow the suffix convention e.g. `ButtonBase`
- [x] All Base component MDX documentation have the base component notification at the top
- [x] Add `mm-` prefix to all classNames
- [x] className is kebab case version of the component name
- [x] Spread base components props and reduce duplication of props when props aren't being changed and remain the same for both variant and base components
- [x] Add component to root `index.js` file in component-library
- [ ] Add locals for any default text I18nContext as default context 
- [ ] Add any "to dos" with a `// TODO:` comment so we can search for them at a later date e.g. blocking components etc
- [x] Add snapshot testing
- [ ] Add pixel values to propType descriptions if we use abstracted prop types that relate to pixel values e.g. `SIZE.MD (32px)`
- [x] Each prop section in the MDX docs should have: a heading, a description, a story and an example code snipped

### CODE REVIEW

File: ui/components/component-library/avatar-with-badge/avatar-with-badge.test.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { BADGE_POSITIONS } from './avatar-with-badge.constants';
 
 describe('AvatarWithBadge', () => {
   it('should render correctly', () => {
-    const { getByTestId } = render(
+    const { getByTestId, container } = render(
       <AvatarWithBadge
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-with-badge/avatar-with-badge.test.js:12]: Should we include a child for this smoke test?

File: ui/components/component-library/index.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ export { AvatarFavicon } from './avatar-favicon';
 export { AvatarIcon, AVATAR_ICON_SIZES } from './avatar-icon';
 export { AvatarNetwork, AVATAR_NETWORK_SIZES } from './avatar-network';
 export { AvatarToken } from './avatar-token';
-export { AvatarWithBadge } from './avatar-with-badge';
+export { AvatarWithBadge, BADGE_POSITIONS } from './avatar-with-badge';
```
Comment (georgewrmarshall) [inline ui/components/component-library/index.js:6]: Hmm our convention is to use the component name in the const but that would make it quite long. I'm `AVATAR_WITH_BADGE_BADGE_POSITIONS` I think just `AVATAR_WITH_BADGE_POSTIONS` could also work. I'm leaning toward keeping our convention at the moment.

**Review by georgewrmarshall — COMMENTED** [review]
Code is looking good! Just need a build to do a visual check

**Review by georgewrmarshall — APPROVED** [review]
LGTM 🚀

**Review by darkwing — APPROVED** [review]

**Comment by brad-decker** [comment]
@georgewrmarshall @NidhiKJha the PR has unchecked checkboxes, are we wanting to move forward with merging?

**Comment by NidhiKJha** [comment]
> @georgewrmarshall @NidhiKJha the PR has unchecked checkboxes, are we wanting to move forward with merging?

Yes, we can merge it! The description had standard conventions for all the housekeeping tickets, some checkboxes are unchecked since they are not apt for the housekeeping of this particular component

**Pull request merged**